changelog/2113.removal.rst

```diff
@@ -0,0 +1 @@
+Removed fallback on certificate ``commonName`` in ``match_hostname()`` function.
```

Comment:
Since this is probably the most breaking change:

 * Can we mention that we're now looking at "Subject Alternative Name"?
 * Can we name drop RFC 2818 from May 2000?
 * Can it be disabled or not?